server/catgenome/src/main/java/com/epam/catgenome/util/motif/MotifSearchIterator.java

```diff
@@ -83,6 +89,7 @@ public MotifSearchIterator(final byte[] seq, final String iupacRegex,
         int position = 0;
         LinkedList<Match> matches = new LinkedList<>();
         while (matcher.find(position)) {
+            Assert.isTrue(matches.size() < maxSearchSize, MESSAGE_TEXT);
```

Comment:
body of this cycle and cycle in `populateNegativeMatches` can be moved to separate method